src/Solvers/col_projection.jl

```diff
@@ -0,0 +1,326 @@
+            )
+        )
+    end
+    # Assuming that max_it is defined in the logger
+    alpha::Float64 = solver.alpha 
+    # We assume the user is using compressors to only decrease dimension
+    n_rows::Int64 = compressor.n_rows
+    n_cols::Int64 = compressor.n_cols
+    initial_size = n_rows 
+    sample_size = n_cols  
+    rows_a, cols_a = size(A)
+    # Allocate the information in the buffer using the types of A and b
+    compressed_mat = typeof(A)(undef, rows_a, sample_size) 
+    residual_vec = typeof(b)(undef, rows_a)
+    # Since sub_solver is applied to compressed matrices use here
+    sub_solver = complete_sub_solver(solver.sub_solver, compressed_mat, residual_vec)
+    mat_view = view(compressed_mat, :, 1:sample_size) 
+    solution_vec = x  
+    update_vec = typeof(x)(undef, n_cols)  
+    return col_projectionRecipe{eltype(A), 
+                        typeof(b), 
+                        typeof(A), 
+                        typeof(mat_view),
+                        typeof(compressor),
+                        typeof(logger),
+                        typeof(error),
+                        typeof(sub_solver)
+                        }(compressor, 
+                        logger, 
+                        error,
+                        sub_solver,
+                        alpha,
+                        compressed_mat,
+                        solution_vec,
+                        update_vec,
+                        mat_view,
+                        residual_vec
+                        )
+end
+
+"""
+    col_proj_update!(solver::col_projectionRecipe)
+
+A function that performs the column projection update when the compression dimension 
+    is one. If ``a`` is the resulting compression of the transpose of the coefficient matrix,
+    and ``r`` is the current residual, then we perform the update:
+
+``x = x - \\alpha S \\frac{\\langle a, r \\rangle}{\\|a\\|_2^2}``,
+
+    where `S` is the compression operator and `a = A S`.
+
+# Arguments
+- `solver::col_projectionRecipe`, the solver information required for performing the update.
+
+# Outputs
+- returns `nothing`
+"""
+function col_proj_update!(solver::col_projectionRecipe)
+    # one-dimensional subarray
+    scaling = solver.alpha * dot(solver.mat_view, solver.residual_vec) 
+    scaling /= dot(solver.mat_view, solver.mat_view)
+    scaling_vec = fill(scaling, 1)
+    # x_new = x_old - alpha * S * update_vec
+    mul!(solver.solution_vec, solver.S, scaling_vec, -1.0, 1.0)
+    # recompute the residual
+    mul!(solver.residual_vec, solver.mat_view, scaling_vec, -1.0, 1.0)
+    return nothing
+end
+
+"""
+    col_proj_update_block!(solver::col_projectionRecipe)
+
+A function that performs the column projection update when the compression dimension 
+    is greater than 1. In the block case, where the compressed matrix is 
+    ``\\tilde A = A S`` and the residual is ``r = b - A x``, we perform the update:
```

Comment:
Residual here is `b-Ax`. In other parts, it is `Ax-b`. The choice of residual needs to be consistent. This choice should be reflected in the code and docstrings.